IdName[] list_id_name